ipam/ipv6Ipam.go

```diff
@@ -0,0 +1,199 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"runtime"
+	"strconv"
+
+	"github.com/Azure/azure-container-networking/common"
+	"github.com/Azure/azure-container-networking/log"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	defaultWindowsKubePath           = `c:\k\`
+	defaultWindowsKubeConfigFilePath = defaultWindowsKubePath + `config`
+	defaultLinuxKubeConfigFilePath   = "/var/lib/kubelet/kubeconfig"
+	nodeSubnetMask                   = "::/120"
+)
+
+type ipv6IpamSource struct {
+	name           string
+	kubeConfigPath string
+	kubeClient     *kubernetes.Clientset
+	kubeNode       *v1.Node
+	sink           addressConfigSink
+}
+
+func newIPv6IpamSource(options map[string]interface{}) (*ipv6IpamSource, error) {
+	var kubeConfigPath string
+	name := options[common.OptEnvironment].(string)
+
+	if runtime.GOOS == windows {
+		kubeConfigPath = defaultWindowsKubeConfigFilePath
+	} else {
+		kubeConfigPath = defaultLinuxKubeConfigFilePath
+	}
+
+	return &ipv6IpamSource{
+		name:           name,
+		kubeConfigPath: kubeConfigPath,
+	}, nil
+}
+
+// Starts the MAS source.
+func (source *ipv6IpamSource) start(sink addressConfigSink) error {
+	source.sink = sink
+
+	return nil
+}
+
+// Stops the MAS source.
+func (source *ipv6IpamSource) stop() {
+	source.sink = nil
+}
+
+func (source *ipv6IpamSource) loadKubernetesConfig() (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", source.kubeConfigPath)
+	client, err := kubernetes.NewForConfig(config)
+	return client, err
+}
+
+func (source *ipv6IpamSource) refresh() error {
+	nodeName, err := os.Hostname()
+
+	if source.kubeClient == nil || source.kubeNode == nil {
+		source.kubeClient, err = source.loadKubernetesConfig()
+		source.kubeNode, err = source.kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	}
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[ipam] Discovered CIDR's %v.", source.kubeNode.Spec.PodCIDRs)
+	if err != nil {
+		return err
+	}
+
+	// Query the list of Kubernetes Pod IPs
+	interfaceIPs, err := retrieveKubernetesPodIPs(source.kubeNode, nodeSubnetMask)
+
+	// Configure the local default address space.
+	local, err := source.sink.newAddressSpace(LocalDefaultAddressSpaceId, LocalScope)
+	if err != nil {
+		return err
+	}
+
+	for _, i := range interfaceIPs.Interfaces {
+		for el, s := range i.IPSubnets {
```

Comment:
what is el  mean? can we rename to something else

Reply:
formerly short for element, but not needed, fixed